src/lib.rs

```diff
@@ -0,0 +1,436 @@
+    pub fn from_call<T>(result: Result<T, KintoError>) -> Result<T, KintoError> {
+        match result {
+            Ok(result) => Ok(result),
+            Err(KintoError::HyperError) => Err(KintoError::HyperError),
+            Err(KintoError::JsonError) => Err(KintoError::JsonError),
+        }
+    }
+}
+
+
+pub struct Client {
+    server_url: String,
+    http_client: client::Client,
+    headers: Headers,
+}
+
+
+impl Client {
+
+    // Create a client
+
+    pub fn new(server_url:String, auth: Authorization<Basic>) -> Client {
+        // Build an SSL connector
+        let ssl = NativeTlsClient::new().unwrap();
+        let connector = HttpsConnector::new(ssl);
+
+        // Build a HTTP Client with TLS support.
+        let client = client::Client::with_connector(connector);
+
+        let mut headers = Headers::new();
+        headers.set(ContentType::json());
+        headers.set(auth.to_owned());
+
+        let url = format!("{}/{}", server_url, "v1");
+
+        Client {
+            server_url: url,
+            headers: headers,
+            http_client: client,
+        }
+    }
+
+    // Utilities
+
+    pub fn server_info(&self) -> Result<JsonValue, KintoError> {
+        let endpoint = format!("{}/", self.server_url);
+        return self.handle_request(Method::Get, endpoint, None);
+    }
+
+    // Buckets
+
+    pub fn get_buckets(&self) -> Result<JsonValue, KintoError> {
+        let endpoint = format!("{}/buckets", self.server_url);
+        return self.handle_request(Method::Get, endpoint, None);
+    }
+
+    pub fn delete_buckets(&self) -> Result<JsonValue, KintoError> {
+        let endpoint = format!("{}/buckets", self.server_url);
+        return self.handle_request(Method::Delete, endpoint, None);
+    }
+
+    pub fn get_bucket(&self, bucket_id: String) -> Result<JsonValue, KintoError> {
+        let endpoint = format!("{}/buckets/{bucket_id}", self.server_url,
+                               bucket_id=bucket_id);
+        return self.handle_request(Method::Get, endpoint, None);
+    }
+
+    pub fn delete_bucket(&self, bucket_id: String) -> Result<JsonValue, KintoError> {
+        let endpoint = format!("{}/buckets/{bucket_id}", self.server_url,
+                               bucket_id=bucket_id);
+        return self.handle_request(Method::Delete, endpoint, None);
+    }
+
+    pub fn create_bucket(&self, data: Option<JsonValue>)
+                         -> Result<JsonValue, KintoError> {
+        return self.create_bucket_with_perms(data, None);
+    }
+
+    pub fn create_bucket_with_perms(&self,
+                                    data: Option<JsonValue>,
+                                    permissions: Option<JsonValue>)
+                                    -> Result<JsonValue, KintoError> {
+
+        let endpoint = format!("{}/buckets", self.server_url);
+        let payload = self.set_payload(data, permissions);
+        return self.handle_request(Method::Post, endpoint, payload.into());
```

Comment:
Instead of `payload.into()` here and elsewhere, I'd find it easier to read as `Some(payload)`.